mava/wrappers/gym.py

```diff
@@ -0,0 +1,392 @@
+        self._env = env
+        self.use_shared_rewards = use_shared_rewards
+        self.add_global_state = add_global_state
+        self.num_agents = len(self._env.action_space)
+        self.num_actions = self._env.action_space[0].n
+
+    def reset(
+        self, seed: Optional[int] = None, options: Optional[dict] = None
+    ) -> Tuple[NDArray, Dict]:
+        if seed is not None:
+            self.env.unwrapped.seed(seed)
+
+        agents_view, info = self._env.reset()
+
+        info = {"actions_mask": self.get_actions_mask(info)}
+        if self.add_global_state:
+            info["global_obs"] = self.get_global_obs(agents_view)
+
+        return np.array(agents_view), info
+
+    def step(self, actions: List) -> Tuple[NDArray, NDArray, NDArray, NDArray, Dict]:
+        agents_view, reward, terminated, truncated, info = self._env.step(actions)
+
+        info = {"actions_mask": self.get_actions_mask(info)}
+        if self.add_global_state:
+            info["global_obs"] = self.get_global_obs(agents_view)
+
+        if self.use_shared_rewards:
+            reward = np.array([np.array(reward).sum()] * self.num_agents)
+        else:
+            reward = np.array(reward)
+
+        return agents_view, reward, terminated, truncated, info
+
+    def get_actions_mask(self, info: Dict) -> NDArray:
+        if "action_mask" in info:
+            return np.array(info["action_mask"])
+        return np.ones((self.num_agents, self.num_actions), dtype=np.float32)
+
+    def get_global_obs(self, obs: NDArray) -> NDArray:
+        global_obs = np.concatenate(obs, axis=0)
+        return np.tile(global_obs, (self.num_agents, 1))
+
+
+class SmacWrapper(GymWrapper):
+    """A wrapper that converts actions step to integers."""
+
+    def step(self, actions: List) -> Tuple[NDArray, NDArray, NDArray, NDArray, Dict]:
+        # Convert actions to integers before passing them to the environment
+        actions = [int(action) for action in actions]
+
+        agents_view, reward, terminated, truncated, info = super().step(actions)
+
+        return agents_view, reward, terminated, truncated, info
+
+    def get_actions_mask(self, info: Dict) -> NDArray:
+        return np.array(self._env.unwrapped.get_avail_actions())
+
+
+class GymRecordEpisodeMetrics(gymnasium.Wrapper):
+    """Record the episode returns and lengths."""
+
+    def __init__(self, env: gymnasium.Env):
+        super().__init__(env)
+        self._env = env
+        self.running_count_episode_return = 0.0
+        self.running_count_episode_length = 0.0
+
+    def reset(
+        self, seed: Optional[int] = None, options: Optional[dict] = None
+    ) -> Tuple[NDArray, Dict]:
+        agents_view, info = self._env.reset(seed, options)
+
+        # Create the metrics dict
+        metrics = {
+            "episode_return": self.running_count_episode_return,
+            "episode_length": self.running_count_episode_length,
+            "is_terminal_step": True,
```

Comment:
Correct me if I'm wrong, but I think this works like the auto-reset wrapper. When `is_terminal_step = True`, it means a new episode started on that step, so it includes the first observation of the new episode along with the final metrics from the previous episode.

Reply:
Good spot Ruan! We do it slightly differently in the anakin version, but it is essentially the same here. 

Here the running counts are only reset after creating the metrics dict, so marking this as a terminal step with the previous steps metrics will work fine. In anakin we always set it to false in reset otherwise we set it to the `done` flag. @Louay-Ben-nessir can we change it to do that for consistence :pray: See [here](https://github.com/instadeepai/Mava/blob/develop/mava/wrappers/episode_metrics.py#L97-L101) for a reference

Reply:
updated! I'd appreciate another pair of eyes of it 👀